Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs

```diff
@@ -0,0 +1,230 @@
+
+            // THE WHAT?
+            // TODO: Need to be able to leave them on the spike to do DoT, see ss13.
+            EntityManager.QueueDeleteEntity(victimUid);
+
+            SoundSystem.Play(Filter.Pvs(uid), component.SpikeSound.GetSound(), uid);
+        }
+
+        private bool TryGetPiece(EntityUid uid, EntityUid user, EntityUid used,
+            KitchenSpikeComponent? component = null, UtensilComponent? utensil = null)
+        {
+            if (!Resolve(uid, ref component) || component.MeatParts == 0)
+                return false;
+
+            // Is using knife
+            if (!Resolve(used, ref utensil) || (utensil.Types & UtensilType.Knife) == 0)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-knife-needed"), uid, Filter.Entities(user));
+                return false;
+            }
+
+            component.MeatParts--;
+
+            if (!string.IsNullOrEmpty(component.MeatPrototype))
+            {
+                var meat = EntityManager.SpawnEntity(component.MeatPrototype, Transform(uid).Coordinates);
+                EntityManager.GetComponent<MetaDataComponent>(meat).EntityName = component.MeatName;
+            }
+
+            if (component.MeatParts != 0)
+            {
+                _popupSystem.PopupEntity(component.MeatSource1p, uid, Filter.Entities(user));
+            }
+            else
+            {
+                UpdateAppearance(uid, null, component);
+                _popupSystem.PopupEntity(component.MeatSource0, uid, Filter.Entities(user));
+            }
+
+            return true;
+        }
+
+        private void UpdateAppearance(EntityUid uid, AppearanceComponent? appearance = null, KitchenSpikeComponent? component = null)
+        {
+            if (!Resolve(uid, ref component, ref appearance, false))
+                return;
+            
+            appearance.SetData(KitchenSpikeVisuals.Status, (component.MeatParts > 0) ? KitchenSpikeStatus.Bloody : KitchenSpikeStatus.Empty);
+        }
+
+        private bool Spikeable(EntityUid uid, EntityUid userUid, EntityUid victimUid,
+            KitchenSpikeComponent? component = null, SharedButcherableComponent? butcherable = null)
+        {
+            if (!Resolve(uid, ref component))
+                return false;
+
+            if (component.MeatParts > 0)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-deny-collect", ("this", uid)), uid, Filter.Entities(userUid));
+                return false;
+            }
+
+            if (!Resolve(victimUid, ref butcherable) || butcherable.MeatPrototype == null)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-deny-butcher", ("victim", victimUid), ("this", uid)), victimUid, Filter.Entities(userUid));
+                return false;
+            }
```

Comment:
Given butcherable can be passed in as null during regular control flow it doesn't make sense to log if it's missing here.